File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    printf("O hai! How much change is owed?\n");
    float change = get_float();
    
    while (change <= 0)
    {
        printf("O hai! How much change is owed?\n");
        change = get_float();
    }

    int changeA = round(change*100);
    int count = 0;
    
    

    while (changeA > 0)
    {
        if (changeA >= 25)
        {
            changeA -= 25;
            count++;
            continue;
            
        } 
        else if (changeA < 25 && changeA > 9)
        {
            changeA -= 10;
            count++;
            continue;
            
        }
        else if (changeA < 10 && changeA > 4)
        {
            changeA -= 5;
            count++;
            continue;
        }
        else 
        {
            changeA -= 1;
            count++;
            continue;
        }
    }

    printf("%i\n", count);
}

File: pset6/greedy.py

def main():
    change = float(input("O hai! How much change is owed? "))
    while change < 0:
        change = float(input("O hai! How much change is owed? "))

    change_alt = round(change * 100)
    count = 0
    while change_alt > 0:
        if change_alt >= 25:
            change_alt -= 25
            count += 1
            continue
        elif change_alt >= 10:
            change_alt -= 10
            count += 1
            continue
        elif change_alt >= 5:
            change_alt -= 5
            count += 1
            continue
        else:
            change_alt -= 1
            count += 1
            continue
    print("{}".format(count))

if __name__ == '__main__':
    main()
